proposals/ref-unsafe-in-iterators-async.md

```diff
@@ -47,7 +47,11 @@ and C# 13 will lift these restrictions as described below.
 [§13.3.1 Blocks > General][blocks-general]:
 
 > ~~It is a compile-time error for an iterator block to contain an unsafe context ([§23.2][unsafe-contexts]).~~
-> An iterator block always defines a safe context, even when its declaration is nested in an unsafe context.
+> An iterator block always defines a safe context, even when its declaration is nested in an unsafe context
+> **unless the iterator method itself is marked with the `unsafe` modifier ([§23.2][unsafe-contexts])**.
```

Comment:
I think we need to clarify what effect, if any, does this have on the signature of the declaration.
For example, is the signature of the method in an `unsafe` context in the following example?
```
unsafe class C
{
    System.Collections.Generic.IEnumerable<int> M() 
    {
        yield return 1;
    }
}
```

Also, consider the following scenario:
```
unsafe class C
{
    System.Collections.Generic.IEnumerable<int> M() 
    {
         unsafe
         {
              { // Does this define a safe context?
                    yield return 1;
              }
         }
    }
}
```

According to the current definition
> A block that contains one or more yield statements is called an iterator block.

Given that, the innermost block is an iterator block Does it define a safe context or not?

I think we also need clarification for properties. For example, is body of the getter in a safe or `unsafe` context in this example?
```
class C
{
     unsafe System.Collections.Generic.IEnumerable<int> P
    {
        get
        {
            yield return 1;
        }
        set
        {}
    }
}
```

Is context safe or unsafe in the setter for this scenario?
```
unsafe class C
{
    System.Collections.Generic.IEnumerable<int> P
    {
        get
        {
            yield return 1;
        }
        set
        {}
    }
}
```